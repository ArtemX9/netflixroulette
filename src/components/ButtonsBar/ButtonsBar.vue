<template>
  <ul v-if="buttons.length" v-on:click="handleClick" class="buttons-bar">
    <li
      v-for="button in buttons"
      :key="button.id"
      :data-id="button.id"
      :class="{ selected: isSelected(button.id) }"
    >
      {{ button.title | capitalize }}
    </li>
  </ul>
</template>

<script>
export default {
  name: "ButtonsBar",
  props: {
    buttons: {
      type: Array,
      required: true
    },
    selectedId: {
      type: String
    }
  },
  methods: {
    handleClick: function({ target }) {
      this.$emit("buttonClick", target.dataset.id);
    },
    isSelected: function(id) {
      return id === this.selectedId;
    }
  },
  filters: {
    capitalize: function(value) {
      if (!value) return "";
      return value.toUpperCase();
    }
  }
};
</script>

<style scoped>
.buttons-bar {
  display: inline-block;
  margin: 0;
  padding: 0;
  font-family: "Trueno Ultra Light", Helvetica, Arial, sans-serif;
}
.buttons-bar li {
  display: inline-block;
  padding: 8px 20px;
  color: white;
  cursor: pointer;
  background-color: rgba(255, 254, 254, 0.2);
}
.buttons-bar li:first-child {
  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;
}

.buttons-bar li:last-child {
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
}

.selected {
  background-color: #f75160 !important;
}
</style>
