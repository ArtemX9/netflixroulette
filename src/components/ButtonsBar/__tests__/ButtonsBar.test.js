import ButtonsBar from "../ButtonsBar.vue";

import { mount } from "@vue/test-utils";

describe("ButtonsBar", () => {
  test("is a Vue instance", () => {
    const wrapper = mount(ButtonsBar, {
      propsData: {
        buttons: [
          {
            id: "some1",
            title: "some1"
          },
          {
            id: "some2",
            title: "some2"
          }
        ]
      }
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders correctly", () => {
    const wrapper = mount(ButtonsBar, {
      propsData: {
        buttons: [
          {
            id: "some1",
            title: "some1"
          },
          {
            id: "some2",
            title: "some2"
          }
        ],
        selectedId: "some2"
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
