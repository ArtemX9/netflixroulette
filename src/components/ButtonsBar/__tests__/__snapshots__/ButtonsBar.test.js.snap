// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsBar renders correctly 1`] = `
<ul
  class="buttons-bar"
>
  <li
    class=""
    data-id="some1"
  >

    SOME1

  </li>
  <li
    class="selected"
    data-id="some2"
  >

    SOME2

  </li>
</ul>
`;
